src/wasm.h

```diff
@@ -1658,6 +1658,12 @@ class Function : public Importable {
     delimiterLocations;
   BinaryLocations::FunctionLocations funcLocation;
 
+  Signature getSig() { return type.getSignature(); }
+  Type getParams() { return getSig().params; }
+  Type getResults() { return getSig().results; }
+  void setParams(Type params) { type = Signature(params, getResults()); }
+  void setResults(Type results) { type = Signature(getParams(), results); }
```

Comment:
the set operations are significantly slower than before, correct? What about the others?

It may be worth doing some benchmarking on this before landing, just to be safe.

Reply:
Yes, constructing or finding the new HeapType is much more expensive than what we previously did, which was to just set a field directly in `Function::sig`. I'm not too worried about that because changing the Function parameters or results is not done except in DAE in the common optimization pass.

`getParams` and `getResults` are also more expensive than before because they now have to chase a pointer (the HeapType ID) rather than directly read the field of the signature in the Function.

I'll benchmark this a bit and see what I find.

Reply:
It doesn't look like this makes an appreciable performance difference when running `wasm-opt -O2` on the poppler binary.

Reply:
Might be worth checking on a large GC testcase too?

Reply:
Ok, looks like this might slow things down by 0.6% (I don't know how statistically significant that is). This seems acceptable to me.

Reply:
sgtm, thanks for checking.